src/python/bot/tasks/fuzz_task.py

```diff
@@ -1873,14 +1881,20 @@ def run(self):
       _track_fuzzer_run_result(self.fuzzer_name, 0, 0,
                                FuzzErrorCode.DATA_BUNDLE_SETUP_FAILED)
       logs.log_error(
-          'Unable to setup data bundle %s.' % fuzzer.data_bundle_name)
+          'Unable to setup data bundle %s.' % self.fuzzer.data_bundle_name)
       return
 
     # For some binaries, we specify trials, which are sets of flags that we only
     # apply some of the time. Adjust APP_ARGS for them if needed.
     trials.setup_additional_args_for_app()
 
-    engine_impl = engine.get(fuzzer.name)
+    # TODO(mbarbella): Remove the environment variable check when the old
+    # blackbox pipeline has been converted to the engine interface.
+    if environment.get_value('USE_BLACKBOX_ENGINE'):
+      engine_impl = engine.get('blackbox')
+    else:
+      engine_impl = engine.get(self.fuzzer.name)
```

Comment:
Let's do it like this to prevent potential issues with never picking the grey box engine if we somehow bungle up our handling of env vars and always keep USE_BLACKBOX_ENGINE set.

```python
engine_impl = engine.get(self.fuzzer.name)
if not engine_impl and environment.get_value('USE_BLACKBOX_ENGINE'):
  engine_impl = engine.get('blackbox')
```